cluster-autoscaler/proposals/pricing.md

```diff
@@ -0,0 +1,177 @@
+ 
+Let’s try something different: 
+```
+NodeUnfitness*(C + X)/(T+X)
+```
+0.1 cpu request:
+
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2  |           4 * (0.095 + 0.016) / (0.003 + 0.016) | 23.36 |
+| N1-standard-8  |         1 * (0.380 + 0.016) / (0.003 + 0.016) | 20.84 |
+| N1-standard-2+GPU |  4 * (0.795 + 0.016) / (0.003 + 0.016) | 168.0 |
+ 
+1.5 cpu request:
+
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2 |           4 * (0.095 + 0.016) / (0.003*15 + 0.016) | 7.24 |
+| N1-standard-8 |          1 * (0.380 + 0.016) / (0.003*15 + 0.016) |  6.49 |
+| N1-standard-2+GPU |   4 * (0.795 + 0.016) / (0.003*15 + 0.016) | 52 |
+ 
+Looks better. So we are able to promote having bigger nodes if needed. However, what if we were to create 50 N1-standard-8 nodes to accommodate 50 x 1.5 cpu pods with strict PodAntiAffinity? Well, in that case we should probably go for n1-standard-2 nodes, however the above formula doesn’t promote that, because it considers the node unfit. So when requesting a larger number of nodes (in a single scale-up) we should probably suppress NodeUnfitness a bit. The suppress function should reduce the effect of NodeUnfitness when there is a good reason to do it. One of the good reasons is that the other option is significantly cheaper. In general the more nodes we are requesting the bigger the price difference can be. And if we are requesting just a signle node
+then this node should be well fitted to the cluster (than to the pod) so that other pods can also use it and the cluster administrator has less types of nodes to focus on. 
+ 
+We are looking for a function suppress(NodeUnfitness, NodeCount) that:
+ 
+* For NodeCount = 1 returns   NodeUnfitness
+* For NodeCount = 2 returns   NodeUnfitness * 0.95
+* For NodeCount = 5 returns   NodeUnfitness * 0.8
+* For NodeCount = 50 returns ~1 which means that the node is perfectly OK for the cluster.
+ 
+A slightly modified sigmoid function has such properties. Lets define
+ 
+```
+suppress(u,n) = (u-1)*(1-math.tanh((n-1)/15.0))+1
+```
+
+Please keep in mind that unfitness is >= 1.
+ 
+Then:
+
+``` 
+suppress(4, 1)=4.000000   == 4 * 1.00
+suppress(4, 2)=3.800296   == 4 * 0.95
+suppress(4, 3)=3.602354   == 4 * 0.90
+suppress(4, 4)=3.407874   == 4 * 0.85
+suppress(4, 5)=3.218439   == 4 * 0.80
+suppress(4,10)=2.388851   == 4 * 0.60
+suppress(4,20)=1.441325   == 4 * 0.36
+suppress(4,50)=1.008712   == 4 * 0.25
+```
+ 
+Exactly what we wanted to have! However, should we need a steeper function we can replace 15 with a smaller number. 
+ 
+So, to summarize, the whole ranking function would be:
+
+``` 
+suppress(NodeUnfitness, NodeCount) * (C + X)/(T + X)
+```
+
+where:
+
+``` 
+suppress(u,n) = (u-1)*(1-math.tanh((n-1)/15.0))+1
+nodeUnFitness = max(preferred_cpu/node_cpu, node_cpu/preferred_cpu)
+```
+
+# Preferred Node
+
+For now we will have a simple, hard-coded values like:
```

Comment:
I think hardcoding is a reasonable approach here, but maybe it would be worth asking more people verify that specific values below make sense.